src/sim/trace-extract/google_trace_simulator.cc

```diff
@@ -891,6 +1300,8 @@ void GoogleTraceSimulator::TaskCompleted(
   JobID_t job_id = JobIDFromString((*td_ptr)->job_id());
   // Remove the task node from the flow graph.
   flow_graph_->TaskCompleted(task_id);
+  // Erase from knowledge base
+  RemoveTaskStats(task_id);
```

Comment:
We now no longer do, although we'll have to monitor memory usage and possibly find ways of compressing the information.